app/models/host_info_providers/puppet_info.rb

```diff
@@ -21,8 +21,23 @@ def puppetclass_parameters
       values = keys.values_hash(host)
 
       klasses = {}
+      klass_error = {}
       host.classes.each do |klass|
-        klasses[klass.name] = smart_class_params_for(klass, key_hash, values)
+        begin
+          klasses[klass.name] = smart_class_params_for(klass, key_hash, values)
+        rescue StandardError => exc
+          klass_error[klass] = exc
+        end
+      end
+      if klass_error.size
+        # FIXME: Is logger in scope under another name?
+        logger = Foreman::Logging.logger('app')
+        klass_error.sort.each do |klass, exc|
+          logger.error("Smart class parameter rendering of `#{klass.name}` failed on #{host.name}: #{exc}")
```

Comment:
Please use `logger.exception` helper method to properly send backtrace into logs.